articles/101_build_tool.md

```diff
@@ -0,0 +1,213 @@
+
+### catkin_make
+
+`catkin_make` is provided by the ROS package `catkin` which contains the build system for ROS 1.
+It has been designed as the successor of `rosbuild` for ROS Fuerte.
+
+The tool invokes CMake only a single time and uses CMake's `include` function to process all packages in a single context.
+While this is an efficient approach since all targets across all packages can be parallelized it comes with significant disadvantages.
+Due to the single context all function names, targets and tests share a single namespace across all packages and on a larger scale this easily leads to collisions.
+The single context is also subject to side effects between the packages and sometimes requires adding additional target dependencies across package boundaries.
+
+### catkin_make_isolated
+
+`catkin_make_isolated` is provided by the ROS package `catkin` which contains the build system for ROS 1.
+It was developed after `catkin_make` to address the problems involved with building multiple packages in a single CMake context.
+
+The tool only supports CMake-based packages and builds each package in topological order using the command sequence common for CMake packages: `cmake`, `make`, `make install`.
+While each package can parallelize the build of its targets the packages are processed sequentially even if they are not (recursive) dependencies of each other.
+
+### ament_tools
+
+`ament_tools` is provided by a standalone Python 3 package used to build ROS 2 packages.
+It was developed to bootstrap the ROS 2 project, is therefore only targeting Python 3, and works on Linux, MacOS and Windows.
+In addition to CMake packages it also supports building Python packages and can infer meta information without requiring an explicit package manifest.
+The tool performs an "isolated" build like `catkin_make_isolated` (one CMake invocation per package) but also parallelizes the build of packages which have no (recursive) dependencies on each other.
+
+### catkin_tools
+
+[catkin_tools](https://catkin-tools.readthedocs.io/) is provided by a standalone Python package used to build ROS 1 packages.
+It was developed after `catkin_make` / `catkin_make_isolated` to build packages in parallel as well as provide significant usability improvements.
+The tool supports building CMake packages and builds them in isolation as well as supports parallelizing the process across packages.
+
+### Naming
+
+The existing build tools are all named by the build system they are supporting.
+In general it should be possible for a build tool to support multiple different build systems.
+Therefore a name for a build tool being derived from a single build system might mislead the users that the tool only works for that specific build system.
+To avoid confusion of the user it would be beneficial if the build tool would have a different name to avoid implying an undesired correlation.
```

Comment:
I would not phrase it as "nice-to-have" since I interpret it to be almost as weak as "not going to happen" :wink: It is clearly not a requirement since any name would "work". So I would call it a "goal" - so stating that this is what we want to do.